internal/services/keyvault/key_vault_certificate_contacts_resource.go

```diff
@@ -158,24 +169,43 @@ func (r KeyVaultCertificateContactsResource) Read() sdk.ResourceFunc {
 			subscriptionResourceId := commonids.NewSubscriptionID(subscriptionId)
 			keyVaultIdRaw, err := vaultClient.KeyVaultIDFromBaseUrl(ctx, subscriptionResourceId, id.KeyVaultBaseUrl)
 			if err != nil {
-				return fmt.Errorf("retrieving resource ID of the Key Vault at URL %s: %+v", id.KeyVaultBaseUrl, err)
+				return fmt.Errorf("retrieving resource ID of the Key Vault from Key Vault Base URL %q in %s: %+v", id.KeyVaultBaseUrl, subscriptionResourceId, err)
 			}
+
 			if keyVaultIdRaw == nil {
-				metadata.Logger.Infof("Unable to determine the Resource ID for the Key Vault at URL %s - removing from state!", id.KeyVaultBaseUrl)
+				metadata.Logger.Infof("[READ] unable to retrieve Key Vault resource ID from Key Vault Base URL %q in %s - removing from state", id.KeyVaultBaseUrl, subscriptionResourceId)
 				return metadata.MarkAsGone(id)
 			}
+
 			keyVaultId, err := commonids.ParseKeyVaultID(*keyVaultIdRaw)
 			if err != nil {
 				return fmt.Errorf("parsing Key Vault ID: %+v", err)
 			}
 
 			existing, err := client.GetCertificateContacts(ctx, id.KeyVaultBaseUrl)
 			if err != nil {
+				// The contacts may or may not have changed but we do not have access to them due
+				// to the key vault public network access has been disabled... should just exit
+				// since we have no access to dataplane... This will preserve the contacts
+				// in the state file so once we do regain access to the dataplane we will be able
+				// to diff the resources contacts with the key vaults contacts...
+				if utils.ResponseWasForbidden(existing.Response) {
+					metadata.Logger.Infof("[READ] unable to enumerate key vault certificate contacts at URL %s in %s - ignore error", id.KeyVaultBaseUrl, subscriptionResourceId)
+					return nil
+				}
+
 				if utils.ResponseWasNotFound(existing.Response) {
-					metadata.Logger.Infof("No Certificate Contacts could be found at %s - removing from state!", id.KeyVaultBaseUrl)
-					return metadata.MarkAsGone(id)
+					metadata.Logger.Infof("[READ] no certificate contacts were returned from key vault URL %s in %s - set empty contact list to state", id.KeyVaultBaseUrl, subscriptionResourceId)
+
+					state := KeyVaultCertificateContactsResourceModel{
+						KeyVaultId: keyVaultId.ID(),
+						Contact:    make([]Contact, 0),
+					}
+
+					return metadata.Encode(&state)
```

Comment:
This then disables one common Terraform behavior that removes non-exist resource from state. Instead, I think we shall reach to the KV service team to ask them to allow creating an empty contact resource (with 0 contact), then things will be much easier.

Reply:
I understand what you are saying, however the core runtime requires all fields to have a value (even if empty). I added this code as a stop gap measure, between the RP and Terraform, that said if we can get the service team to agree to allow empty key vault contacts that would be ideal, however I am not sure how successful we will be in convincing they to do so, which is why I introduced this wrapper code as it is sort of equivalent.